### ISSUE TITLE
Support adaptive sizing

### ISSUE THREAD

**Issue by brendanlong** [jangernert/FeedReader#771]
The Librem 5 phone is going to run GNOME apps, so it would be cool if FeedReader ran on it. Apparently they created some new widgets that will let us show/hide panels based on screen size. This might be nice on desktop too, since we could scale to only take up half the screen.

http://bytesgnomeschozo.blogspot.com/2018/10/librem-5-gnome-332.html
https://developer.puri.sm/Apps/GNOME/Gtk+.html

We can probably do something similar to what I did in Feedbin, where we hide the least-useful current panel and add back buttons:

https://www.brendanlong.com/fluid-css-design-and-feedbins-mobile-interface.html

i.e. if we don't have space for all three panels and we're looking at an article, hide the feed list. If we not looking at an article, hide the empty article panel. If we can only fit one panel, show one at a time with back buttons.

**Comment by brendanlong** [comment]
Yeah I'm getting a Librem5 so ideally this will work on it. Although I'm not sure if our CPU usage will be tolerable :\

**Comment by jangernert** [comment]
From what I've seen the the widgets in [libhandy](https://source.puri.sm/Librem5/libhandy/tree/master/src) are still somewhat limited. The best solution would be to write a custom UI on top of the news_flash core, possibly sharing lots of things with the other Gtk+ UI(s).
Is there a genreal HIG for the Librem5 that isn't likely to change soon?
But yeah, I wouldn't suggest working on a port for FeedReader.

**Comment by Kekun** [comment]
Looking at the current layout from FeedReader, which is basically the same as Geary, something that could work is what I did for my Geary prototype a year ago:
- having a HdyLeaflet in the titlebar and the content of the window, in sync.
- having a HdyLeaflet nested in the left side of the window's content leaflet.

That way you would have three adaptive modes:
- large, as it currently is
- medium, showing either the two left panels and their header bar or the right panel and its header bar
- narrow, showing only one of the panels at a time and its corresponding header bar

**Comment by thibaultamartin** [comment]
Purism [released lib handy v0.0.10](https://puri.sm/posts/the-new-libhandy-0-0-10/) a few days ago (June 17th).

I had a quick chat with Purism's @Kekun who's actually behind libhandy, who told me that the API should be stable despite the apparently low version number.

If you intend to adapt FeedReader for the Librem 5 you can start using libhandy today, you shouldn't waste time rewriting it as the retrocompatibility should not break when releasing v1.0 :)

Thanks for the great software, and I can't wait to see it on running on a fully Libre phone!
